### PR TITLE
fix issue #46: ehloCmd method is not run when Mail Server reply "ESMTP" is not all uppercase characters

### PR DESCRIPTION
**Pull request by alexlehm**
moved isEsmtpSupported to Utils class, added unit test for the Utils method and a test in HeloTest
regenerated docs and generated js files

### CODE REVIEW

File: vertx-mail-client/src/main/java/io/vertx/ext/mail/impl/Utils.java [hunk 1]
```diff
@@ -22,13 +22,19 @@
 import java.util.ArrayList;
```
Comment (cescoffier) [inline vertx-mail-client/src/main/java/io/vertx/ext/mail/impl/Utils.java]: license header missing

**Comment by cescoffier** [comment]
Everything fine except the missing header.

File: vertx-mail-client/src/main/java/io/vertx/ext/mail/impl/Utils.java (hunk 1 above)
Comment (alexlehm) [inline vertx-mail-client/src/main/java/io/vertx/ext/mail/impl/Utils.java]: I assume the license header is the same as in the other files, are you sure its wrong?

Comment (cescoffier) [inline vertx-mail-client/src/main/java/io/vertx/ext/mail/impl/Utils.java]: It's not a new file, so it's fine.

**Comment by alexlehm** [comment]
changed the comments in the Utils class

**Pull request merged**